core/src/main/java/com/linecorp/armeria/common/SystemPropertyFlagsProvider.java

```diff
@@ -0,0 +1,446 @@
+
+    @Override
+    public String headerValueCacheSpec() {
+        return getNormalized("headerValueCacheSpec");
+    }
+
+    @Override
+    public List<String> cachedHeaders() {
+        final String val = getNormalized("cachedHeaders");
+        if (val == null) {
+            return null;
+        }
+        return CSV_SPLITTER.splitToList(val);
+    }
+
+    @Override
+    public String fileServiceCacheSpec() {
+        return getNormalized("fileServiceCacheSpec");
+    }
+
+    @Override
+    public String dnsCacheSpec() {
+        return getNormalized("dnsCacheSpec");
+    }
+
+    @Override
+    public Predicate<InetAddress> preferredIpV4Addresses() {
+        final String val = getNormalized("preferredIpV4Addresses");
+        if (val == null) {
+            return null;
+        }
+        final List<Predicate<InetAddress>> preferredIpV4Addresses =
+                CSV_SPLITTER.splitToList(val)
+                            .stream()
+                            .map(cidr -> {
+                                try {
+                                    return InetAddressPredicates.ofCidr(cidr);
+                                } catch (Exception e) {
+                                    logger.warn("Failed to parse a preferred IPv4: {}", cidr);
+                                }
+                                return null;
+                            })
+                            .filter(Objects::nonNull)
+                            .collect(Collectors.toList());
+        switch (preferredIpV4Addresses.size()) {
+            case 0:
+                return null;
+            case 1:
+                return preferredIpV4Addresses.get(0);
+            default:
+                return inetAddress -> {
+                    for (Predicate<InetAddress> preferredIpV4Addr : preferredIpV4Addresses) {
+                        if (preferredIpV4Addr.test(inetAddress)) {
+                            return true;
+                        }
+                    }
+                    return false;
+                };
+        }
+    }
+
+    @Override
+    public Boolean useJdkDnsResolver() {
+        return getBoolean("useJdkDnsResolver");
+    }
+
+    @Override
+    public Boolean reportBlockedEventLoop() {
+        return getBoolean("reportBlockedEventLoop");
+    }
+
+    @Override
+    public Boolean validateHeaders() {
+        return getBoolean("validateHeaders");
+    }
+
+    @Override
+    public Boolean tlsAllowUnsafeCiphers() {
+        return getBoolean("tlsAllowUnsafeCiphers");
+    }
+
+    @Override
+    public Set<TransientServiceOption> transientServiceOptions() {
+        final String val = getNormalized("transientServiceOptions");
+        if (val == null) {
+            return null;
+        }
+        return Sets.immutableEnumSet(
+                Streams.stream(CSV_SPLITTER.split(val))
+                       .map(feature -> TransientServiceOption.valueOf(
+                               Ascii.toUpperCase(feature)))
```

Comment:
nit: Join two lines?
```suggestion
                       .map(feature -> TransientServiceOption.valueOf(Ascii.toUpperCase(feature)))
```